js/modules/modules.go

```diff
@@ -42,6 +42,10 @@ type VU interface {
 	// Context return the context.Context about the current VU
 	Context() context.Context
 
+	// Events allows subscribing to global k6 execution events, such as Init and
+	// Exit, and to local (per-VU) events, such as IterStart and IterEnd.
+	Events() common.Events
```

Comment:
One concern that I have in my mind (but unfortunately not a proposal to resolve it) is that we expose the events (both `global` and `local`) under the VU, where logically and from the context here should be only the `local` (VU-related), but access to the global (Test Run-related) seems to be somewhere else. Have you considered this? :thinking:

Reply:
I had a comment somewhere about this, but I must've deleted it.

The thing is that there's no way to pass "global" objects to JS modules. The only access to k6 they have is via this `VU` interface, since effectively every module instance is tied to a single VU.

So I agree with you; global objects don't belong here. I think there should be another way for modules to receive things like this and consolidated options that's not tied to a single VU, but to the global module itself. I experimented with passing these in some `ModuleState` object in the module constructor (e.g. `browser.New()`), but options are not consolidated at that point yet. We could pass these static global objects that way, though, and reuse the `Init` event to pass options, for example, but it's just an idea, and we haven't discussed it yet. Or we could expose another global API modules could access from... somewhere.

So, yeah, this would need some design discussion. Currently I see no way around this, other than removing the global event system, and using only the local per-VU one, but that has other complications...

Reply:
Also, let me stress that I'm not considering this as a blocker as long as we explicitly (with the comment maybe) mention that this API could change since we see a "separation of concerns" issue and users know that they could use "global" events keeping in mind that this API could change.

Reply:
Can we put a Note that this is experimental API.

Also we probably should drop it from `RegisterCallback` :thinking: